WordPress/WordPressShareExtension/ShareExtractor.swift

```diff
@@ -452,34 +452,35 @@ private struct URLExtractor: TypeBasedExtensionContentExtractor {
             return item
         }
 
-        let converter = Down(markdownString: md)
-        guard let html = try? converter.toHTML(.safe) else {
-            return item
-        }
+        return handleMarkdown(md: md, item: item)
+    }
 
+    private func handleMarkdown(md content: String, item: ExtractedItem? = nil) -> ExtractedItem? {
+        var md = content
         var result: ExtractedItem
         if let item = item {
             result = item
         } else {
             result = ExtractedItem()
         }
-        result.importedText = html
 
-        return result
-    }
+        // If the first line is formatted as a heading, use it as the title
+        var lines = md.components(separatedBy: .newlines)
+        if lines.count > 1, lines[0].first == "#" {
```

Comment:
Just a style nit (maybe) → any thoughts on breaking this logic out into a separate func?